providers/implementations/rands/seed_src_jitter.c

```diff
@@ -194,20 +194,20 @@ static int jitter_generate(void *vseed, unsigned char *out, size_t outlen,
         return 0;
     }
 
-    if (adin != NULL && adin_len > 0) {
-        if (!ossl_rand_pool_add(pool, adin, adin_len, 0)) {
-            ERR_raise(ERR_LIB_PROV, ERR_R_RAND_LIB);
-            ossl_rand_pool_free(pool);
-            return 0;
-        }
-    }
-
     /* Get entropy from jitter entropy library. */
     entropy_available = ossl_prov_acquire_entropy_from_jitter(s, pool);
 
     if (entropy_available > 0)
         memcpy(out, ossl_rand_pool_buffer(pool), ossl_rand_pool_length(pool));
 
+    if (adin != NULL && adin_len > 0) {
+        size_t i;
+
+        /* xor the additional data into the output */
+        for (i = 0; i < adin_len; ++i)
+            out[i % outlen] ^= adin[i];
```

Comment:
would it make sense to do something like?
```
for (i = 0; i < MAX(addin_len, outlen); i++)
   out[i % outlen] ^= adin[i % addin_len];
```

just a question as I said code looks correct.

Reply:
No, it is not necessary to xor over all the entropy output buffer.